src/blockrelay/blockrelay_common.h

```diff
@@ -20,8 +21,18 @@ struct CThinTypeBlockInFlight
     int64_t nRequestTime;
     bool fReceived;
     const std::string thinType;
+    bool operator==(const CThinTypeBlockInFlight &b) { return ((hash == b.hash) && (thinType == b.thinType)); }
+    friend bool operator<(const CThinTypeBlockInFlight &a, const CThinTypeBlockInFlight &b)
+    {
+        if ((a.hash == b.hash) && (a.thinType == b.thinType))
+        {
+            return false;
+        }
+        return (a.hash < b.hash);
+    }
```

Comment:
shouldn't this just return false?  why the need for comparing hash values here?

Reply:
a std::set<T> requires <T> to have a < operator. std::set uses the < operator as a comparator for its internal sorting function. it is determined that two objects are equial when comparator(a,b) and comparator(b,a) both return false. i compare the hashes as a way to sort. the hash seemed to be the best datamember out of the 4 to use for sorting due to its uniqueness. 

but this comment made me realize it needs a slight tweak though. so i have adjusted it.

Reply:
I was thinking of something else...looks ok.